File: qlty-cli/tests/lang/ruby/basic.in/nested_control.rb
def not_nested(foo, bar)
    if (foo == 'cat' and bar == 'dog') or (foo == 'dog' and bar == 'cat')
        puts 'Got a cat and a dog!'
    else
        puts 'Got nothing'
    end
end

def f0
    if bar
        if baz
            if qux
                if quux
                    puts 'Not deeply nested enough!'
                end
            end
        end
    end
end

def f1
    if bar
        if baz
            if qux
                if quux
                    puts 'Deeply nested!'
                end
            end
        end
    end
end

def f2(foo)
    case foo
    when 1
        'bar1'
    when 2
        'bar2'
    when 3
        'bar3'
    when 4
        'bar4'
    when 5
        'bar5'
    when 6
        'bar6'
    when 7
        'bar7'
    when 8
        'bar8'
    when 9
        'bar9'
    when 10
        'bar10'
    else
        raise ArgumentError, 'Invalid foo value'
    end
end

File: qlty-cli/tests/lang/java/basic.in/NestedControl.java
class Main {
    public static void notNested(String foo, String bar) {
        if (("cat".equals(foo) && "dog".equals(bar)) || ("dog".equals(foo) && "cat".equals(bar))) {
            System.out.println("Got a cat and a dog!");
        } else {
            System.out.println("Got nothing");
        }
    }

    public static void f0(boolean bar, boolean baz, boolean qux, boolean quux) {
        if (bar) {
            if (baz) {
                if (qux) {
                    if (quux) {
                        System.out.println("Not deeply nested enough!");
                    }
                }
            }
        }
    }

    public static void f1(boolean bar, boolean baz, boolean qux, boolean quux) {
        if (bar) {
            if (baz) {
                if (qux) {
                    if (quux) {
                        System.out.println("Deeply nested!");
                    }
                }
            }
        }
    }

    public static String f2(int foo) {
        switch (foo) {
            case 1:
                return "bar1";
            case 2:
                return "bar2";
            case 3:
                return "bar3";
            case 4:
                return "bar4";
            case 5:
                return "bar5";
            case 6:
                return "bar6";
            case 7:
                return "bar7";
            case 8:
                return "bar8";
            case 9:
                return "bar9";
            case 10:
                return "bar10";
            default:
                throw new IllegalArgumentException("Invalid foo value");
        }
    }
}
